### PR TITLE
Add CAPTCHA

### PR DESCRIPTION
**Pull request by dpoirier**
There are two places we want to require a CAPTCHA:

    1) On a user's first vote 
    2) At account registration

Otherwise we shouldn't require it.

### CODE REVIEW

File: opendebates/settings.py [hunk 1]
```diff
@@ -260,3 +261,7 @@ def __contains__(self, item):
 DBBACKUP_STORAGE_OPTIONS = {'location': '.'}
 DBBACKUP_FILENAME_TEMPLATE = 'local/{datetime}.{extension}'
 DBBACKUP_SEND_EMAIL = False
+
+# Dummy settings for test etc
+NORECAPTCHA_SITE_KEY = ''
+NORECAPTCHA_SECRET_KEY = ''
```
Comment (vkurup) [inline opendebates/settings.py]: Wondering what the least intrusive way to do this for local development is... 

Hmm, looks like Google provides a fake sitekey and secret: https://developers.google.com/recaptcha/docs/faq#id-like-to-run-automated-test-with-recaptcha-v2-how-should-i-do (2nd question) That seems to work.

File: opendebates/templates/opendebates/vote.html [hunk 2]
```diff
@@ -129,46 +129,7 @@
          class="btn btn-default">Create an Account</a>
       {% endif %}
     {% endcomment %}
-    <form action="{% url 'vote' id=idea.id %}" method="POST" class="hidden form-inline">
```
Comment (vkurup) [inline opendebates/templates/opendebates/vote.html]: +1 for removing unused code.

File: opendebates/views.py [hunk 3]
```diff
@@ -211,7 +213,7 @@ def vote(request, id):
         msg = 'That email is registered. Please login and try again.'
         if request.is_ajax():
             return HttpResponse(
-                json.dumps({"status": "400", "errors": {'email': msg}}),
+                json.dumps({"status": "400", "errors": {'email': [msg]}}),
```
Comment (vkurup) [inline opendebates/views.py]: thanks for the fix.

File: opendebates/settings.py (hunk 1 above)
Comment (dpoirier) [inline opendebates/settings.py]: The norecaptcha library provides a shortcut too, which you can see in use in some of the tests.

**Comment by vkurup** [comment]
Seems to work. I get the captcha on my first vote, but no my second, and also on the registration form.

:+1: after adding fake creds for local development

**Comment by ejucovy** [comment]
Do we need to worry about performance impacts here, since it means first-time vote requests (which could potentially come in big bursts on launch day and during other big announcements) will have to wait on a server-to-server validation response from Google?  What happens to the pool of available workers if responses from Recaptcha start lagging or hanging?  Or is there a timeout baked in somewhere in the validation API call?  (Maybe I'm just fretting pointlessly and the answer is "Google's servers are the least of our performance worries..")

**Comment by dpoirier** [comment]
Performance is definitely a concern. But I believe most of the work is done between the user's browser and Google, which results in a token being passed to us. We still have to call Google to verify that the token is valid, but hopefully that's a fast API - it would need to be since other high-traffic sites wouldn't want to use recaptcha otherwise.

**Pull request merged**